scan_struct.go

```diff
@@ -50,7 +103,9 @@ func ScanStruct(r pgx.Row, dest interface{}) error {
 	}
 
 	// scan into the struct field pointers and append to our results
-	return r.Scan(values...)
+	rows.Scan(values...)
```

Comment:
Kind of counter-intuitively, you can still safely use the `return r.Scan(values...)` here. All you _really_ need from the `*connRows` is the `rows.FieldDescriptions()`, everything else can still be handled by the `r.Scan` just like it used to be.

Additionally, if they start exposing the `FieldDescriptions` in the `pgx.Row` interface, you only need to change one line, and delete all the reflection stuff.

Reply:
> everything else can still be handled by the r.Scan just like it used to be

it can not - to call the `rows.FieldDescriptions()` I need to call `rows.Next()` first as it reads metadata from the first row, and the same call happens in the `r.Scan()`, that means it will either return `pgx.ErrNoRows` if there was only row in the result (most likely) or will read the second row as result that is not what it should do.

Reply:
:cry: this is probably why the `pgx.Row` interface does not expose the `FieldDescriptions` method.